data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/stream/BinlogEventListener.java

```diff
@@ -59,17 +63,20 @@ public class BinlogEventListener implements BinaryLogClient.EventListener {
 
     private final StreamRecordConverter recordConverter;
     private final BinaryLogClient binaryLogClient;
-    private final BufferAccumulator<Record<Event>> bufferAccumulator;
+    private final Buffer<Record<Event>> buffer;
     private final List<String> tableNames;
     private final String s3Prefix;
     private final boolean isAcknowledgmentsEnabled;
     private final PluginMetrics pluginMetrics;
     private final List<Event> pipelineEvents;
     private final StreamCheckpointManager streamCheckpointManager;
+    private final ExecutorService executorService;
```

Comment:
It may be good to give this a name indicating what we use it for. e.g. `binlogEventExecutorService`.

Reply:
Updated.